thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/alert/scheme/DetectionAlertScheme.java

```diff
@@ -80,4 +80,17 @@ public static BaseNotificationContent buildNotificationContent(Properties alertS
     LOG.info("Using " + content.getClass().getSimpleName() + " to render the template.");
     return content;
   }
+
+  /**
+   * Fail the alert task if unable to notify owner. However, in case of dimensions recipient alerter,
+   * do not fail the alert if a subset of recipients are invalid.
+   */
+  void handleAlertFailure(int size, IllegalArgumentException e) {
+    if (this.result.getResult().size() == 1) {
+      throw e;
+    } else {
+      LOG.warn("Skipping! Found illegal arguments while sending {} anomalies for alert {}." + " Exception message: ",
```

Comment:
shall we log which recipient failed?

Reply:
It should be part of the exception message.

Reply:
I see